pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -700,6 +709,35 @@ protected synchronized boolean trySendMessagesToConsumers(ReadType readType, Lis
         return true;
     }
 
+    public Consumer getNextConsumer(List<Entry> entries) {
+        if (!redeliveryTracker.hasRedeliveredEntry(entries)){
```

Comment:
should this code run only in case of "InMemoryAndPreventCycleFilterRedeliveryTracker" ?
this way users that don't need that feature (don't have filters) won't be impacted

Reply:
Good idea, already fixed